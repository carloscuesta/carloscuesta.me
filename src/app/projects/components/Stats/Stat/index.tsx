type Props = { label: string; value: string; unit?: string }

const Stat = (props: Props) => (
  <div className="flex-1">
    <article className="rounded-lg border border-solid border-neutral-100 px-4 py-8 dark:border-neutral-800">
      <p className="pb-2 opacity-80">{props.label}</p>
      <p className="text-xl font-extrabold">
        {props.value}
        {props.unit && (
          <span className="text-sm font-medium opacity-50">{` ${props.unit}`}</span>
        )}
      </p>
    </article>
  </div>
)

export default Stat
