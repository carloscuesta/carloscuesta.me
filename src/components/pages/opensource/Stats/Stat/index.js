// @flow
import React, { type Element } from 'react'

import styles from './styles.module.css'

type Props = { label: string, value: number }

const Stat = (props: Props): Element<'div'> => (
  <div className='col-xs-12 col-sm-6 col-md-4'>
    <article className={styles.card}>
      <p className={styles.label}>{props.label}</p>
      <strong className={styles.value}>
        {props.value.toLocaleString()}
      </strong>
    </article>
  </div>
)

export default Stat
