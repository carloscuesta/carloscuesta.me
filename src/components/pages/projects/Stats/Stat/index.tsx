type Props = { label: string, value: string, unit?: string }

const Stat = (props: Props) => (
  <div className='flex-1'>
    <article className="border-neutral-100 border-solid border px-4 py-8 rounded-lg dark:border-neutral-800">
      <p className="pb-2 opacity-80">{props.label}</p>
      <p className="font-extrabold text-xl">
        {props.value}
        {props.unit  && (
          <span className="font-medium text-sm opacity-50">{` ${props.unit}`}</span>
        )}
      </p>
    </article>
  </div>
)

export default Stat
