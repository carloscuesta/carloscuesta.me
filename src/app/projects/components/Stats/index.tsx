import Stat from './Stat'

type Props = {
  followers: string
  packageDownloads: string
  stars: string
}

const Stats = (props: Props) => (
  <section className="grid gap-4 sm:grid-flow-col sm:grid-cols-3">
    <Stat label="GitHub Stars" value={props.stars} />
    <Stat label="GitHub Followers" value={props.followers} />
    <Stat label="NPM Downloads" value={props.packageDownloads} unit="/ year" />
  </section>
)

export default Stats
