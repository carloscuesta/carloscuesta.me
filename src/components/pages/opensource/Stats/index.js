// @flow
import React, { type Element } from 'react'

import SectionTitle from 'src/components/shared/SectionTitle'
import Stat from './Stat'

type Props = {
  followers: number,
  packageDownloads: number,
  repositories: number,
  stars: number
}

const Stats = (props: Props): Element<'section'> => (
  <section>
    <SectionTitle title='Stats' />

    <div className='row'>
      <Stat label='GitHub Stars' value={props.stars} />
      <Stat label='GitHub Followers' value={props.followers} />
      <Stat label='Repositories' value={props.repositories} />
      <Stat label='NPM Downloads (Last year)' value={props.packageDownloads} />
    </div>
  </section>
)

export default Stats
