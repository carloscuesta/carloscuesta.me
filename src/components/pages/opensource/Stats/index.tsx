import SectionTitle from 'src/components/shared/SectionTitle'
import Stat from './Stat'

type Props = {
  followers: number,
  packageDownloads: number,
  stars: number
}

const Stats = (props: Props) => (
  <section>
    <SectionTitle title='Stats' />

    <div className='row'>
      <Stat label='GitHub Stars' value={props.stars} />
      <Stat label='GitHub Followers' value={props.followers} />
      <Stat label='NPM Downloads (Last year)' value={props.packageDownloads} />
    </div>
  </section>
)

export default Stats
