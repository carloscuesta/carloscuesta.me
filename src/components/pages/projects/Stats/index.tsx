import Stat from './Stat'

type Props = {
  followers: string,
  packageDownloads: string,
  stars: string
}

const Stats = (props: Props) => (
  <section className='grid sm:grid-flow-col sm:grid-cols-3 gap-4'>
    <Stat label='GitHub Stars' value={props.stars} />
    <Stat label='GitHub Followers' value={props.followers} />
    <Stat label='NPM Downloads' value={props.packageDownloads} unit='/ year' />
  </section>
)

export default Stats
