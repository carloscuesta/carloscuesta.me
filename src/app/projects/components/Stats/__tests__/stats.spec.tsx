import { render } from '@testing-library/react'

import Stats from '../index'
import * as stubs from './stubs'

describe('Stats', () => {
  it('should match Stats component', () => {
    const { container } = render(<Stats {...stubs.props} />)

    expect(container).toMatchSnapshot()
  })
})
