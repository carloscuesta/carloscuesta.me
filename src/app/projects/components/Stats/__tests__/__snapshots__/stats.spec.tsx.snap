// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats should match Stats component 1`] = `
<section
  className="grid gap-4 sm:grid-flow-col sm:grid-cols-3"
>
  <div
    className="flex-1"
  >
    <article
      className="rounded-lg border border-solid border-neutral-100 px-4 py-8 dark:border-neutral-800"
    >
      <p
        className="pb-2 opacity-80"
      >
        GitHub Stars
      </p>
      <p
        className="text-xl font-extrabold"
      >
        1,000
      </p>
    </article>
  </div>
  <div
    className="flex-1"
  >
    <article
      className="rounded-lg border border-solid border-neutral-100 px-4 py-8 dark:border-neutral-800"
    >
      <p
        className="pb-2 opacity-80"
      >
        GitHub Followers
      </p>
      <p
        className="text-xl font-extrabold"
      >
        100
      </p>
    </article>
  </div>
  <div
    className="flex-1"
  >
    <article
      className="rounded-lg border border-solid border-neutral-100 px-4 py-8 dark:border-neutral-800"
    >
      <p
        className="pb-2 opacity-80"
      >
        NPM Downloads
      </p>
      <p
        className="text-xl font-extrabold"
      >
        100,000
        <span
          className="text-sm font-medium opacity-50"
        >
           / year
        </span>
      </p>
    </article>
  </div>
</section>
`;
